types/compare.go

```diff
@@ -35,6 +37,50 @@ func CompareUint64(x, y uint64) int {
 	return 1
 }
 
+//VecComparUU returns []int64 comparing the []uint64 x to []uint64 y
+func VecComparUU(x, y []uint64, res []int64) {
+	n := len(x)
+	for i := 0; i < n; i++ {
+		if x[i] < y[i] {
+			res[i] = -1
+		} else if x[i] == y[i] {
+			res[i] = 0
+		} else {
+			res[i] = 1
+		}
+	}
+}
+
+//VecCompareII returns []int64 comparing the []int64 x to []int64 y
+func VecCompareII(x, y, res []int64) {
```

Comment:
`VecCompareII(x, y, res []int64, n)` ?

Reply:
I had refined it. Thank you ~

Reply:
I think we can keep the former implementation, it's clearer, and the performance has no difference.
``` 
$ GO111MODULE=on go test -bench BenchmarkResOfLT -run=none -count=5
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkResOfLT-4       1709767               728 ns/op
BenchmarkResOfLT-4       1446705               723 ns/op
BenchmarkResOfLT-4       1616919               782 ns/op
BenchmarkResOfLT-4       1597010               713 ns/op
BenchmarkResOfLT-4       1656105               713 ns/op
BenchmarkResOfLT2-4      1698483               700 ns/op
BenchmarkResOfLT2-4      1715144               720 ns/op
BenchmarkResOfLT2-4      1610162               766 ns/op
BenchmarkResOfLT2-4      1582326               730 ns/op
BenchmarkResOfLT2-4      1701380               710 ns/op
```

``` go
func initData() (x []int64) {
	x = make([]int64, 1024)
	for i :=0 ; i < 1024; i++{
		x[i] = int64(i)
	}
	return
}

func vecResOfLT2(res []int64) {
	n := len(res)
	for i := 0; i < n; i++ {
		if res[i] < 0 {
			res[i] = 1
		} else {
			res[i] = 0
		}
	}
}


func BenchmarkResOfLT(b *testing.B) {
	x := initData()
	b.ResetTimer()
	for i := 0; i < b.N; i++{
		vecResOfLT(1024, x)
	}
}

func BenchmarkResOfLT2(b *testing.B){
	x := initData()
	b.ResetTimer()
	for i := 0; i < b.N; i++{
		vecResOfLT2(x)
	}
}

```

Reply:
I have rolled to the previous version